### ISSUE TITLE
asyncPartWayThrough should move proceed existing loads to next state.

### ISSUE THREAD

**Issue by justinbmeyer** [ModuleLoader/es6-module-loader#305]
First discussed here https://github.com/systemjs/systemjs/pull/347, there is a problem with `System.define` calls defining a load that has already been added to `loader.loads`.

@guybedford Any suggestions on the best way of creating a test for this?  The only idea I have is to:

1. Begin an import of "master" module that has a "slave" dependency.
2. Create a locate (perhaps fetch) hook to delay the "master" from resolving its "slave" dependency.
3. Within that hook, `System.define` the "slave" module.
4. Resolve the "master"'s locate (or fetch) hook.

Thoughts?

**Comment by guybedford** [comment]
@justinbmeyer this sounds sensible. I can't dedicate time myself to this until next week or later unfortunately, but if you come up with a test please do share it.

**Comment by alexisabril** [comment]
Noting a quick/dirty conceptual fix for this issue: https://github.com/bitovi/es6-module-loader/commit/697717fc9415864b009d5fb302e6c43b93e61534

**Comment by alexisabril** [comment]
I added a PR for our initial fix and will be submitting a test shortly. The issue is currently visible by using https://github.com/bitovi/systemjs/commit/9b403cab390201a81d04d049d582a8288c480329 in SystemJS(described in https://github.com/systemjs/systemjs/pull/347).

What we've observed so far is within `proceedToTranslate`, the `LoadSucceeded` section is called twice. Initially, it is called with the `loaded` status, however the `System.define` `slave` is called again with a `linked` status(set in `updateLinkSetOnLoad`. This throws the assertion error as `linked` is unexpected by this point.

From: https://github.com/ModuleLoader/es6-module-loader/blob/master/src/loader.js#L307
```
console.assert(load.status == 'loading', 'is loading');

load.status = 'loaded';

var linkSets = load.linkSets.concat([]);
for (var i = 0, l = linkSets.length; i < l; i++)
  updateLinkSetOnLoad(linkSets[i], load);
```
